### DISCUSSION TITLE
Inheritance or conditional destination type

### DISCUSSION THREAD

**Discussion by ggalejandro** [nartc/mapper#382] [Q&A]
I'm trying to write a mapper for a class wich has an attribute with mixed typed, all of them with inheritance relation, but I cant't find a way to do that.

Trying to provide a simpliffied example model
```
enum UserType {
 TYPEONE: "TYPEONE";
 TYPETWO: "TYPETWO";
}

class User {
 name: string;
 tye: UserType;
}

class UserOne extends User{
extra: string 
}

class UserTwo extends User{
 differentExtra: number:  
}

class Group {
 users: (UserOne|UserTwo)[]
}
```

I want to map groups in some way the mapper could instantiate UserOne or UserTwo based on type. 
In an ideal way I'd like to do an UserMap, that dynamically return one type or another, but I have not found any function to do that. 
I tried using mapDefer() in combination with mapWith() at both levels, but at User level, it does not override the full destination, and at group level, mapDefer does not work with array members, only with the array itself. 

Any help? Thanks in advance.

**Comment by nartc** [comment]
Can I see your map configuration?

**Comment by Bugslive** [comment]
Hi all,
I have currently the same interrogation: I would be able to specify which mapping to use based on a property "type" and I didn't find out how to do it for array items ? Is there a way to do it with current API or would it require a new mapping configuration ?
